tests/components/renault/test_binary_sensor.py

```diff
@@ -0,0 +1,198 @@
+"""Tests for Renault binary sensors."""
+from unittest.mock import patch
+
+import pytest
+from renault_api.kamereon import exceptions
+
+from homeassistant.components.binary_sensor import DOMAIN as BINARY_SENSOR_DOMAIN
+from homeassistant.const import (
+    ATTR_IDENTIFIERS,
+    ATTR_MANUFACTURER,
+    ATTR_MODEL,
+    ATTR_NAME,
+    ATTR_SW_VERSION,
+    STATE_OFF,
+    STATE_UNAVAILABLE,
+)
+from homeassistant.setup import async_setup_component
+
+from . import (
+    setup_renault_integration_vehicle,
+    setup_renault_integration_vehicle_with_no_data,
+    setup_renault_integration_vehicle_with_side_effect,
+)
+from .const import MOCK_VEHICLES
+
+from tests.common import mock_device_registry, mock_registry
+
+
+@pytest.mark.parametrize("vehicle_type", MOCK_VEHICLES.keys())
+async def test_binary_sensors(hass, vehicle_type):
+    """Test for Renault binary sensors."""
+    await async_setup_component(hass, "persistent_notification", {})
+    entity_registry = mock_registry(hass)
+    device_registry = mock_device_registry(hass)
+
+    with patch("homeassistant.components.renault.PLATFORMS", [BINARY_SENSOR_DOMAIN]):
+        await setup_renault_integration_vehicle(hass, vehicle_type)
+        await hass.async_block_till_done()
+
+    if vehicle_type == "captur_fuel":
+        # fuel vehicles do not have any binary sensors
+        assert len(device_registry.devices) == 0
```

Comment:
This doesn't seem correct. We expect one device even if there are no entities for a platform. It's even part of our fixture.

Reply:
I wasn't aware that "we expect one device even if there are no entities for a platform", so I guess this is something that needs to be plugged in, and renders the "if" discussion redundant.

I'll start work on this right away.

Reply:
If there are no entities at all for an integration there will be no entities that registers a device via `device_info`. But here that doesn't seem to be the case. I think the reason is that we patch the platforms constant to only set up the binary sensor platform. I think that is not the correct approach since it makes our asserting incorrect.

Side note: I'd expect this integration to create one device per car regardless of what entities are exposed.